pyqtgraph/parametertree/parameterTypes/basetypes.py

```diff
@@ -274,22 +274,28 @@ def __init__(self, *args, **kargs):
 
     def _interpretValue(self, v):
         typ = self.opts['type']
+
         def _missing_interp(v):
             # Assume raw interpretation
             return v
             # Or:
             # raise TypeError(f'No interpreter found for type {typ}')
+
         interpreter = getattr(builtins, typ, _missing_interp)
         return interpreter(v)
 
+
 class GroupParameterItem(ParameterItem):
     """
     Group parameters are used mainly as a generic parent item that holds (and groups!) a set
     of child parameters. It also provides a simple mechanism for displaying a button or combo
     that can be used to add new parameters to the group.
     """
+
     def __init__(self, param, depth):
         ParameterItem.__init__(self, param, depth)
+        # prevent 0 depth item's font from growing on every call of updateDepth
+        self.fontPointSize = partial(self.font(0).pointSize)
```

Comment:
I have no experience with this. 
If this defines a `fontPointSize` method for WidgetParameterItem, maybe the boring, verbose way of going through `def fontPointSize` might be more readable?

Reply:
In this specific case, this is basically the same as `self.fontPointSize = lambda: self.font(0).pointSize()` but is marginally faster. 
I didn't define a method because I didn't want to add any methods to the object that weren't necessary elsewhere in the code. 

I think this is pretty readable as is but I'll change this into a property instead. 

```py
    @ property
    def fontPointSize(self):
        return self.font(0).pointSize()
```

Reply:
Oh, I find that much easier to read :)
"I will give you this if asked for fontPointSize()"

Last question:
This is not solved by simply overriding / providing a matching `pointSize()` method here?

Reply:
I don't think I follow..  If you're asking "Why does this need to be a property, wouldn't a method work just the same?" the answer is yes, it would work but ` font.setPointSize(self.fontPointSize() + 1)` looks weird compared to `                font.setPointSize(self.fontPointSize + 1)`

Reply:
This is nitpicky and not a large concern. Overall, we *typically* don't reach into other objects to grab their properties. Even though we are not Java, it *seems* that we more commonly go through accessor methods. 

I think my viewpoint is
- if this is used internally only (as in your example), then do we even need it over accessing `self.font(0).pointSize()` directly where applicable?
- if this provides a regular interface for other parts of the code, going through a normal method definition is the easiest to find and document. And it clearly sells "use this to get this information". This would be particularly true if you are basically reporting the effective pointsize for this `ParameterItem`. Other objects do that through `pointSize()`, wouldn't it be nice to match that?

Reply:
> if this is used internally only (as in your example), then do we even need it over accessing self.font(0).pointSize() directly where applicable?

Actually, now that I think about it, the method or lambda won't work either. 

```
# we define the font as
font = self.font(0/1)
# we increment based on the current size
font.setPointSize(self.font(0/1).pointSize() + 1)  # this will grow when palette is changed. 
# same goes for anything that calls the method of self.font().pointSize()
# to get the current size, as its increased every time we set the font
self.setFont(0/1,font)
```

If we set the values in the `__init__` 

```

self.fontPointSize = partial(self.font(0).pointSize) # set only once when class initialized 
self.fontPointSize = self.font(0).pointSize          # set only once when class initialized
# we increment with the same pointSize every time updateDepth is run
font.setPointSize(self.fontPointSize()+1) 
# when the font is set it will always use size 14(or whatever the actual size is)
```

Reply:
So the issue then is that ```self.font(0)``` returns a *copy of* (instead of a reference to) a ```QFont```?
From the Qt docs:
```c++
QFont QTreeWidgetItem::font(int column) const
```